script.js

```diff
@@ -1,85 +1,103 @@
-		let button = createButton(mode.name)
-		button.size(BUTTON_WIDTH, BUTTON_HEIGHT)
-		button.style('font-size', '24px')
-		button.position(
-			(width - BUTTON_WIDTH) / 2,
+  file_selector.addEventListener("change", (event) => {
+    addNewCsvPhraseList(event);
+  });
+
+  // Set the frame rate
+  frameRate(fr);
+
+  // Create buttons for mode selection
+  for (let i = 0; i < modes.length; i++) {
+    let mode = modes[i];
+    let button = createButton(mode.name);
+    button.size(BUTTON_WIDTH, BUTTON_HEIGHT);
+    button.style("font-size", "24px");
+    button.position(
+      (width - BUTTON_WIDTH) / 2,
       //700 was originally "height", but it messes up on smaller viewports. Not entirely sure why.
-			(700 - modes.length * (BUTTON_HEIGHT + BUTTON_GAP) * i) / 2 - BUTTON_GAP
-		)
-		button.mousePressed(() => {
-			currentMode = modes[i]
-			modeSelectButtons.forEach((b) => b.hide())
-			state = PLAY_STATE
-		})
-		button.hide()
-		modeSelectButtons.push(button)
-	}
-  
+      (700 - modes.length * (BUTTON_HEIGHT + BUTTON_GAP) * i) / 2 - BUTTON_GAP
+    );
+    button.mousePressed(() => {
+      currentMode = modes[i];
+      modeSelectButtons.forEach((b) => b.hide());
+      state = PLAY_STATE;
+    });
+    button.hide();
+    modeSelectButtons.push(button);
+  }
 }
 
 function draw() {
-	switch (state) {
-		case SELECT_MODE_STATE:
-			showMainScreen()
-			break
-		case PLAY_STATE:
-			currentMode.run()
-			break
-	}
+  switch (state) {
+    case SELECT_MODE_STATE:
+      showMainScreen();
+      break;
+    case PLAY_STATE:
+      currentMode.render();
+      break;
+  }
 }
 
+// TODO encapsulate the main screen into a proper class that extends Screen
 function showMainScreen() {
-	background(50, 150, 150)
-	fill(0, 50, 50)
-	textAlign(CENTER, CENTER)
-	textSize(30)
-	text('Char Char Bang!', width / 2, 60)
-	textSize(30)
-	fill(150, 150, 150)
-	text('Please select a game mode', width / 2, 120)
-	modeSelectButtons.forEach((b) => b.show())
+  background(50, 150, 150);
+  fill(0, 50, 50);
+  textAlign(CENTER, CENTER);
+  textSize(30);
+  text("Char Char Bang!", width / 2, 60);
+  textSize(30);
+  fill(150, 150, 150);
+  text("Please select a game mode", width / 2, 120);
+  modeSelectButtons.forEach((b) => b.show());
 }
 
 function keyPressed() {
-	// Allow the user to reset the game via a special button
-	if (key === 'F2') state = SELECT_MODE_STATE
-	// Otherwise we ignore the shift key and pass the input to the game for processing.
-	else if (key !== 'Shift' && currentMode) currentMode.processKeyInput(key)
+  // Allow the user to reset the game via a special button
+  if (key === "F2") state = SELECT_MODE_STATE;
+  // Otherwise we ignore the shift key and pass the input to the game for processing.
+  else if (key !== "Shift" && currentMode) currentMode.processKeyInput(key);
+}
+
+function mousePressed() {
+  let paintedButtons = currentMode.paintedButtons();
+  paintedButtons.forEach((b) => b.checkForClick(mouseX, mouseY));
```

Comment:
Since currentMode is not initialized in the menu, it generates an error on every click. It might be good to add a check for undefined, or a check for the current state, which may be more versatile.